### ISSUE TITLE
BestPractices: Warning messages do not fail the tests

### ISSUE THREAD

**Issue by artem-lunarg** [KhronosGroup/Vulkan-ValidationLayers#8030]
The warning messages are detected when surrounded by `SetDesiredFailureMsg`/`VerifyFound` in the negative tests, but they do not fail the positive tests, and also do fail the negative tests where the api usage is assumed to be correct but in fact violates some BP rule.

Warning error reporting can be enabled with:

```
void VkBestPracticesLayerTest::InitBestPracticesFramework() {
    m_errorMonitor->ExpectSuccess(kErrorBit | kWarningBit);
    ..
```
This starts reporting issues in the existing BP tests, and they should be fixed.

Here are the few categories of errors I noticed:
- debug utils are enabled as part of the test framework, and BP complains about it. One option is to ignore this for all the tests except the ones that check this specific use case.
- some tests violate BP rules, should be just fixed
- one special case is that there is a warning against using ALL_COMMANDS_BIT stage, and all stage masks are checked against it. That might be a questionable advice, because there are situations when ALL_COMMANDS_BIT is the only choice. We can discuss if this check should be removed.
